File: Book/Fortran/Fig_12.17_ompVecPi.f90
          PROGRAM MAIN
          IMPLICIT NONE

          INTEGER :: i
          INTEGER, PARAMETER :: num_steps = 100000
          REAL :: x, pi, sum, step

          sum = 0.0

          step = 1.0/num_steps

!$OMP parallel SIMD private(x) reduction(+:sum)
          DO i = 1, num_steps
             x = (i - 0.5) * step
             sum = sum + 4.0 / (1.0 + x * x)
          ENDDO
!$OMP end parallel SIMD

          pi = step * sum
          print *, "pi=", pi

          END PROGRAM MAIN 

File: Book/C/Fig_12.17_ompVecPi.c
#include <omp.h>
#include <stdio.h>
static long num_steps = 100000;
float step;
int main ()
{
   int i;
   float x, pi, sum = 0.0;

   step = 1.0f / (double) num_steps;

   #pragma omp simd private(x) reduction(+:sum)
      for (i = 0; i < num_steps; i++) {
         x = (i + 0.5f) * step;
         sum += 4.0f / (1.0f + x * x);
      }

   pi = step * sum;
   printf("pi=%lf \n", pi);
}	  
